common/txmgr/priority_queue.go

```diff
@@ -0,0 +1,101 @@
+package txmgr
+
+import (
+	"container/heap"
+	"sync"
+
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	txmgrtypes "github.com/smartcontractkit/chainlink/v2/common/txmgr/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+// TxPriorityQueue is a priority queue of transactions prioritized by creation time. The oldest transaction is at the front of the queue.
+type TxPriorityQueue[
+	CHAIN_ID types.ID,
+	ADDR, TX_HASH, BLOCK_HASH types.Hashable,
+	R txmgrtypes.ChainReceipt[TX_HASH, BLOCK_HASH],
+	SEQ types.Sequence,
+	FEE feetypes.Fee,
+] struct {
+	sync.RWMutex
+	*txmgrtypes.PriorityQueue[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]
```

Comment:
I'm wondering a little why we need a TxPriorityQueue instead of just accessing the PriorityQueue directly. I see that the mutex is in the TxPriorityQueue, though we could move that logic to the PriorityQueue to ensure that anyone who uses it would have a thread-safe PriorityQueue. And there are two higher level functions (RemoveTxById) and PruneByTxId that wouldn't entirely make sense to be built into the PriorityQueue. But the FindById function is in the PriorityQueue interface, and that's the main thing this functions rely on, so in theory, you could just put all this in PriorityQueue and get rid of TxPriorityQueue right? Unless I'm missing something.